ui/src/main/resources/edu/wpi/grip/ui/templates/java/BasicStep.vm

```diff
@@ -0,0 +1,28 @@
+		//input
+#foreach($input in $step.getInputs())
+#if( $input.mutable() )##Is Mutable
+#if( $input.hasValue() )##Has a set value
+		$input.baseType() $input.name()$c = $input.value();
+#else##Has Connected Socket
+		$input.baseType() $input.name()$c = ${input.value()}.get().$input.baseType()Value();
+#end##end Has a set value
+#else##Not Mutable
+#if ( $input.type() == "FlipCode" )
+		$input.type() $input.name()$c = FlipCode.$input.value();
+#else
+		$input.type() $input.name()$c = $input.value();
+#end
+#end##end Is Mutable
+#end
+		//output
+#foreach($output in $step.getOutputs())
+#if( $output.mutable() )
+		#MutOf($output.type()) $output.name() = new #MutOf($output.type())();
+#else
+		$output.type() $output.name() = new $output.type()();
+#end
+#end
```

Comment:
Ah - never mind the tangent I mentioned; I see that name uniqueness for outputs is handled in TPipeline::set.